### ISSUE TITLE
minimal example of wrong conversion compared to libsixel

### ISSUE THREAD

**Issue by joseluis** [mkrueger/icy_sixel#2]
Hi @mkrueger, I've made a minimal example with an unexpected wrong conversion compared to libsixel.

This is the minimal original png image:
![map8](https://github.com/mkrueger/icy_sixel/assets/5832068/7559d82c-41d8-42ef-ad09-7e91c619e1b5)

This screenshot shows the compared result:
![output](https://github.com/mkrueger/icy_sixel/assets/5832068/72b36022-bdf8-47c3-a289-bb9c44164ea9)

This is the libsixel script:
```
#!/bin/sh
# NOTE: img2sixel installed with:
# $ sudo apt install libsixel-bin
#
# $ img2sixel --version
# img2sixel 1.10.3
#
# configured with:
#   libcurl: no
#   libpng: no
#   libjpeg: no
#   gdk-pixbuf2: no
#   GD: no

img2sixel -o map8-img2sixel.six map8.png
```

This is the rust script:
```rust
#!/usr/bin/env -S cargo +nightly -Zscript
---
[package]
edition = "2021"
[dependencies]
icy_sixel = "0.1.2"
image = { version = "0.24", features = ["png"] }
---
//
// // ↑ Above script header uses Rust nightly
// // ↓ Below script header uses https://crates.io/crates/rust-script
//
// #!/usr/bin/env -S rust-script -c
// //! ```cargo
// //! [package]
// //! edition = "2021"
// //! [dependencies]
// //! icy_sixel = "0.1.2"
// //! image = { version = "0.24", features = ["png"] }
// //! ```

use icy_sixel::{
    sixel_string, DiffusionMethod, MethodForLargest, MethodForRep, PixelFormat, Quality,
};
use image::{codecs::png::PngDecoder, ImageDecoder};
use std::{
    fs::File,
    io::{BufReader, Write},
    path::Path,
};

fn main() {
    let file = File::open("map8.png").unwrap();
    let reader = BufReader::new(file);
    let decoder = PngDecoder::new(reader).unwrap();
    let (width, height) = decoder.dimensions();
    dbg![decoder.color_type()];

    let mut img = vec![0; decoder.total_bytes() as usize];
    decoder.read_image(&mut img).unwrap();

    // I've played with a lot of the settings but none seem to fix the output
    let sixel_data = sixel_string(
        &img,
        width as i32,
        height as i32,
        PixelFormat::RGB888,
        DiffusionMethod::None,
        MethodForLargest::Auto,
        MethodForRep::Auto,
        Quality::AUTO,
    )
    .unwrap();

    let mut file = File::create(Path::new("map8-icy_sixel.six")).unwrap();
    file.write_all(sixel_data.as_bytes()).unwrap();
}
```

**Comment by mkrueger** [comment]
I only visit this site rarely - it's safer to contact me via mail.

That issue should now be fixed with

https://github.com/mkrueger/icy_sixel/commit/053d22cbc8832b7d0dbea94354f0a5497c14e065

However I needed to change the input format:

```
let mut img_rgb888 = Vec::new();
    for i in 0..img.len() / 4 {
        img_rgb888.extend(&img[i * 4..i * 4 + 3]);
    }
```

Then it's RGB888 - but there were some issues in icy_sixel which are now hopefully fixed. I should invest some more time into this project but I'm atm too busy elsewhere.  I made some subtle mistakes in the c->rust conversion.

First task should be to port the unit tests :).
